src/people.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import { AppContext } from './AppContext';
+
+export const People = React.memo(
+  () => {
+    const {
+      personName,
+      setPersonName,
+      applyQuery,
+      filteredPeople,
+      handleClick,
+    } = useContext(AppContext);
+
+    return (
+      <div className="dropdown is-active">
+        <div className="dropdown-trigger">
+          <input
+            type="search"
+            placeholder="Enter a part of the name"
+            className="input"
+            value={personName}
+            onChange={(event) => {
+              setPersonName(event.target.value);
+              applyQuery(event.target.value);
+            }}
```

Comment:
Pls move the function inside `onChange` to separate const and use as callback here